File: create-icons.js
// Script Node.js simples para criar ícones básicos
// Execute: node create-icons.js
// Requer: npm install canvas (ou use a versão HTML)

const fs = require('fs');
const path = require('path');

// Criar diretório se não existir
const iconsDir = path.join(__dirname, 'assets', 'icons');
if (!fs.existsSync(iconsDir)) {
  fs.mkdirSync(iconsDir, { recursive: true });
}

// Criar ícones SVG simples (que podem ser convertidos depois)
const sizes = [16, 48, 128];

sizes.forEach(size => {
  const svg = `<?xml version="1.0" encoding="UTF-8"?>
<svg width="${size}" height="${size}" xmlns="http://www.w3.org/2000/svg">
  <rect width="${size}" height="${size}" fill="#25D366" rx="${size * 0.2}"/>
  <path d="M ${size * 0.2} ${size * 0.2} 
           L ${size * 0.8} ${size * 0.2} 
           Q ${size * 0.9} ${size * 0.2} ${size * 0.9} ${size * 0.3}
           L ${size * 0.9} ${size * 0.6}
           Q ${size * 0.9} ${size * 0.7} ${size * 0.8} ${size * 0.7}
           L ${size * 0.5} ${size * 0.7}
           L ${size * 0.35} ${size * 0.85}
           L ${size * 0.4} ${size * 0.7}
           L ${size * 0.2} ${size * 0.7}
           Q ${size * 0.1} ${size * 0.7} ${size * 0.1} ${size * 0.6}
           L ${size * 0.1} ${size * 0.3}
           Q ${size * 0.1} ${size * 0.2} ${size * 0.2} ${size * 0.2}
           Z" fill="white"/>
</svg>`;
  
  fs.writeFileSync(path.join(iconsDir, `icon${size}.svg`), svg);
  console.log(`Criado: icon${size}.svg`);
});

console.log('\nÍcones SVG criados!');
console.log('Para converter para PNG, você pode:');
console.log('1. Usar um conversor online (ex: cloudconvert.com)');
console.log('2. Abrir generate-icons.html no navegador');
console.log('3. Ou usar ImageMagick: magick convert icon16.svg icon16.png');

File: create_icons.py
#!/usr/bin/env python3
"""
Script para criar ícones PNG básicos para a extensão
Execute: python create_icons.py
Requer: pip install Pillow
"""

try:
    from PIL import Image, ImageDraw
except ImportError:
    print("Erro: Pillow não instalado. Execute: pip install Pillow")
    exit(1)

import os

# Criar diretório se não existir
icons_dir = os.path.join('assets', 'icons')
os.makedirs(icons_dir, exist_ok=True)

def create_icon(size):
    """Cria um ícone PNG simples com o tamanho especificado"""
    # Criar imagem com fundo verde WhatsApp
    img = Image.new('RGB', (size, size), color='#25D366')
    draw = ImageDraw.Draw(img)
    
    # Desenhar balão de mensagem branco
    margin = int(size * 0.2)
    width = size - (margin * 2)
    height = int((size - (margin * 2)) * 0.7)
    
    # Corpo do balão (retângulo arredondado)
    x1, y1 = margin, margin
    x2, y2 = margin + width, margin + height
    radius = int(size * 0.1)
    
    # Desenhar retângulo arredondado
    draw.rounded_rectangle([x1, y1, x2, y2], radius=radius, fill='white')
    
    # Desenhar cauda do balão (triângulo)
    tail_points = [
        (margin + int(width * 0.3), margin + height),
        (margin + int(width * 0.2), margin + int(height * 1.3)),
        (margin + int(width * 0.4), margin + int(height * 1.1))
    ]
    draw.polygon(tail_points, fill='white')
    
    return img

# Criar ícones em diferentes tamanhos
sizes = [16, 48, 128]
for size in sizes:
    icon = create_icon(size)
    filename = os.path.join(icons_dir, f'icon{size}.png')
    icon.save(filename, 'PNG')
    print(f'Criado: {filename}')

print(f'\nTodos os ícones criados em {icons_dir}/')

